peps/pep-0791.rst

```diff
@@ -0,0 +1,165 @@
+`Open Issues <Open Issues_>`_ section.  New functions are not part of the
+proposal.
+
+
+Backwards Compatibility
+=======================
+
+As aliases in :external+py3.14:mod:`math` will be kept for indefinite time
+(their use would be discouraged), there are no anticipated code breaks.
+
+
+Reference Implementation
+========================
+
+https://github.com/python/cpython/pull/133909
+
+
+Open Issues
+===========
+
+Module name
+-----------
+
+Chosen name seems consistent with other domain-specific mathematical module:
+:external+py3.14:mod:`cmath` (for complex numbers).
+
+There is already an ``imath`` project on PyPI, but only with two releases, with
+the most recent one 4 years ago.  Its repository is no longer accessible.
+The `Imath <https://github.com/AcademySoftwareFoundation/Imath>`_ C++ library
+include Python bindings with same name.
+
+`Polling shows <https://discuss.python.org/t/91337/35>`_ ``intmath`` as another
+popular name.  The argument made was that the normal mathematical spelling of
+the imaginary unit is ``i``, which makes imath ambiguous.  It also has no conflict
+with any PyPI module.  On the other hand, ``intmath`` may be confused with
+interval math or numerical integration.
+
+Other proposed names include ``ntheory`` (like SymPy's submodule),
+``integermath`` and ``imaths``.
+
+
+Module scope and possible extensions
+------------------------------------
+
+Unless we can just provide bindings to some well supported mathematical library
+like the GMP, the module scope should be limited.  For example, no primality
+testing and factorization.
+
+There are possible additions, among proposed in the initial discussion thread
+(see also [5]_):
+
+* ``c_div()`` --- for integer ceiling divide, see [2]_, [3]_.
+* ``gcdext()`` --- to solve linear Diophantine equation in two variables (the
+  :external+py3.14:class:`int` implementation actually include extended
+  Euclidean algorithm)
+* ``isqrt_rem()`` --- to return both integer square root and a remainder (if
+  integer isn't a perfect square)
+* ``ilog()`` --- integer logarithm, currently :external+py3.14:func:`math.log`
+  has a special handling for integer arguments.  It's unique (wrt other module
+  functions) and not documented so far, see [4]_
+* ``fibonacci()``.
+
+
+Rejected ideas
+==============
+
+There was a brief discussion about exposing :external+py3.14:func:`math.isqrt`
+as ``imath.sqrt`` in the same way that :external+py3.14:func:`cmath.sqrt` is
+the complex version of :external+py3.14:func:`math.sqrt`.  However, ``isqrt``
+is ultimately a different function: it is the floor of the square root.  It
+would be confusing to give it the same name (under a different module).
+
+
+Acknowledgements
+================
+
+Thanks to Tim Peters for reviving the idea of the :external+py3.14:mod:`math`
+splitting.  Thanks to Neil Girdhar for substantial improvements of
+the initial draft.
+
+
+Footnotes
```

Comment:
Where possible, please replace these footnotes with inline, direct links. That way it saves people a click.